controllers/crd/crd_controller.go

```diff
@@ -99,5 +103,12 @@ func (r *ReconcileCRD) Reconcile(ctx context.Context, req reconcile.Request) (re
 		}
 	}
 
+	// If Perses CRDs are now available, restart to register Perses controller and cache the new GVKs.
+	if hcoutil.IsPersesAvailable(ctx, r.client) {
```

Comment:
If something is changed in the CRD, this code will cause restart even if the CRD were already installed.

WDYT about adding a new boolean field in the reconciler struct, to hold the result of `hcoutil.IsPersesAvailable` on boot time (when the reconciler is created), then check it here; for example:


```suggestion
	if !r.presesAvalableOnBoot && hcoutil.IsPersesAvailable(ctx, r.client) {
```

This will make sure we are doing the restart only once, only when perses is installed.